network/kademlia.go

```diff
@@ -902,6 +903,68 @@ func (k *Kademlia) string() string {
 	return "\n" + strings.Join(rows, "\n")
 }
 
+func (k *Kademlia) MarshalJSON() (data []byte, err error) {
```

Comment:
Even this method just returns the representation of Kademlia, it contains logic that should be tested. I think that testing would be beneficial here.